app/controllers/api/v2/courses_controller.rb

```diff
@@ -168,6 +169,11 @@ def update_subjects
         return if subject_ids.nil?
 
         @course.subjects = Subject.where(id: subject_ids)
+
+        subject_ids.map(&:to_i).each_with_index do |id, index|
+          @course.course_subjects.select { |cs| cs.subject_id == id }.first.priority = index
+        end
+
```

Comment:
I would make this a bit more concise, and just add the course subjects directly, since we have to expose that part already.

```ruby
        @course.course_subjects = subject_ids.map.with_index do |subject_id, idx|
          CourseSubject.create(priority: idx, subject_id: subject_id)
        end
```

Reply:
We don't need this for with the `:before_add` it seems :eyes:

Reply:
Though they do have to be added in this way to keep the ordering (Subject.where is ordered by ID)

Reply:
I thought it might be useful here just to re-enforce the ordering, and I think it's possible the hook could trigger a SQL request. But those issues aside, ya, we may not need/want it here.